joomla/joomla-cms
# [4.0] Fix mail test

Pull Request for Issue #18779 .

### Summary of Changes

Fixed the mail test in configuration for https://github.com/joomla/joomla-cms/pull/18779

The change of this PR includes:

- Fixed CSRF for Ajax
- Fixed Ajax error handler
- Fixed JS get value from new switch button. (Since it has 2 inputs with same name, we must add `:checked` selector)
- Use `JsonInput` to get test mail information.
- Use Namespaces.
- Add missing translations.

### Testing Instructions

Click `Send Test Mail`, should display correct message (Success or Error, not CSRF invalid message).

![p-2017-11-23-001](https://user-images.githubusercontent.com/1639206/33170420-35b30dfe-d083-11e7-8719-5436dda69630.jpg)

![p-2017-11-23-002](https://user-images.githubusercontent.com/1639206/33170421-35e1f326-d083-11e7-8319-0de0ed06360c.jpg)
